modules/edu.gemini.web.server.common/src/test/scala/edu/gemini/web/common/http4s/StaticRoutesSpec.scala

```diff
@@ -47,47 +48,47 @@ class StaticRoutesSpec extends FlatSpec with Matchers with EitherValues with Uri
   "StaticRoutes index(true)" should
     "return a generated index(true) with production files on production mode" in {
       val service = new StaticRoutes(index(false), false, builtAtMillis).service
-      service.apply(Request(uri = uri("/"))).unsafePerformSync.orNotFound.status should equal (Status.Ok)
-      service.apply(Request(uri = uri("/"))).unsafePerformSync.orNotFound.headers should contain (`Content-Type`(`text/html`, Charset.`UTF-8`))
-      service.apply(Request(uri = uri("/"))).unsafePerformSync.orNotFound.headers should contain (`Cache-Control`(NonEmptyList(`no-cache`())))
-      val body = service.apply(Request(uri = uri("/"))).unsafePerformSync.orNotFound.body.runLast.unsafePerformSync.map(_.decodeUtf8).fold("")(_.right.value)
-      body should include regex ".*jsdeps\\.min\\.(\\d*)\\.js.*"
-      body should include regex ".*app-opt\\.(\\d*)\\.js.*"
+      service.apply(Request(uri = uri("/"))).value.map(_.map(_.status)).unsafeRunSync should contain(Status.Ok)
+      service.apply(Request(uri = uri("/"))).value.map(_.map(_.headers)).unsafeRunSync.getOrElse(Headers.empty) should contain (`Content-Type`(`text/html`, Charset.`UTF-8`))
+      service.apply(Request(uri = uri("/"))).value.map(_.map(_.headers)).unsafeRunSync.getOrElse(Headers.empty) should contain (`Cache-Control`(NonEmptyList.of(`no-cache`())))
+      val body = service.apply(Request(uri = uri("/"))).value.map(_.map(_.body.through(text.utf8Decode).compile.foldMonoid)).unsafeRunSync.map(_.unsafeRunSync())
+      body.getOrElse("") should include regex ".*jsdeps\\.min\\.(\\d*)\\.js.*"
+      body.getOrElse("") should include regex ".*app-opt\\.(\\d*)\\.js.*"
     }
     it should "return a generated index(true) with development files on dev mode" in {
       val service = new StaticRoutes(index(true), true, builtAtMillis).service
-      service.apply(Request(uri = uri("/"))).unsafePerformSync.orNotFound.status should equal (Status.Ok)
-      service.apply(Request(uri = uri("/"))).unsafePerformSync.orNotFound.headers should contain (`Content-Type`(`text/html`, Charset.`UTF-8`))
-      service.apply(Request(uri = uri("/"))).unsafePerformSync.orNotFound.headers should contain (`Cache-Control`(NonEmptyList(`no-cache`())))
-      val body = service.apply(Request(uri = uri("/"))).unsafePerformSync.orNotFound.body.runLast.unsafePerformSync.map(_.decodeUtf8).fold("")(_.right.value)
-      body should include regex ".*jsdeps\\.js.*"
-      body should include regex ".*app*\\.js.*"
+      service.apply(Request(uri = uri("/"))).value.map(_.map(_.status)).unsafeRunSync should contain(Status.Ok)
```

Comment:
FYI `fa.map(_.map(f))` can be written `Nested(fa).map(f).value` which is sometimes nicer. What you have here is fine though.

Reply:
Will `Nested` work on more than two levels?

Reply:
No, it just goes down one level.

```scala
@ Nested(Nested(Option(Option(Option(1))))).map(_ + 1).value.value 
res13: Option[Option[Option[Int]]] = Some(Some(Some(2)))
```